### ISSUE TITLE
Random primitive shape for 0-D arrays is always 1

### ISSUE THREAD

**Issue by taless474** [STEllAR-GROUP/phylanx#827]
Having
```py
@Phylanx
def random_uniform(shape, minval, maxval, dtype=None, seed=None):
	return random(shape, list("uniform", minval, maxval))
print(random_uniform(10, 0.0, 1.0))
```
one would expect 10 random generated numbers in [0, 1.) while only 1 number is generated.

**Comment by scheherzade** [comment]
@hkaiser The same problem exists for 1d, 2d, and 3d cases, if the input is passed as an array the current implementation generates a random array with the same shape as the input.
```
cout(random([2,4]))
```
returns an array like this:
```
[-1.19211, -0.374654]
```
instead of something like this:
```
[[1.09431, -1.166, 1.37005, -1.67725], [0.798906, 0.167385, -0.352471, -0.321691]]
```

**Comment by taless474** [comment]
I should add that `random` sometimes has `nil` somewhere in its output. In physl:
```
random(make_list(5,10), list("binomial", 3, .5))
```
produces:
```
[[2, 2, 3nil,
 0, 2, 1, 1, 1, 1, 2], [2, 2, 2, 0, 1, 1, 2, 1, 3, 2], [3, 2, 2, 1, 0, 1, 2, 2, 2, 0], [2, 0, 1, 2, 1, 3, 3, 1, 0, 2], [2, 1, 1, 2, 2, 2, 2, 2, 1, 1]]
```

**Comment by taless474** [comment]
Actuallu, `nil` problem has occured in other cases, too. Therefore, it should not be caused by random.

**Comment by hkaiser** [comment]
@taless474 do you have a reproducing test case for the `nil` problem?

**Comment by taless474** [comment]
Sure. In physl
```console
cout(softmax([[1, 2, 33], [42, 5, 6]]))
```
results
```log
[[1.26642e-nil1
4, 3.44248e-14, 1], [1, 8.53305e-17, 2.31952e-16]]
```

**Comment by hkaiser** [comment]
@taless474 I'm not able to reproduce this. Any other hints you may give to see this?

**Comment by hkaiser** [comment]
@taless474 I think I know what's happening. Do you use the `physl` interpreter with the `-p/--print` command line option? If yes, then the interpreter will print the overall result of the evaluation of the given expression (`nil` in this case), which will get interspersed with the output generated from the evaluation itself. We probably should use `hpx::cout` instead of `std::cout` in the interpreter to avoid this.

**Comment by taless474** [comment]
@hkaiser, good catch. I am using physl interpreter with `--print`.

**Comment by hkaiser** [comment]
@taless474 this should be fixed by #854

**Comment by taless474** [comment]
It is. Thanks
